// Copyright 2025 The Ray Authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//  http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.


syntax = "proto3";

import "google/protobuf/timestamp.proto";
import "src/ray/protobuf/runtime_env_common.proto";
import "src/ray/protobuf/common.proto";

package ray.rpc.events;

// Message containing the execution information of a driver job. It can be used to
// capture the full state transition history.
//
// For static information associated with this event, see DriverJobDefinitionEvent.
message DriverJobExecutionEvent {
  enum State {
    UNSPECIFIED = 0;
    CREATED = 1;
    FAILURE = 2;
    SUCCESS = 3;
  }

  message StateTimestamp {
    State state = 1;
    google.protobuf.Timestamp timestamp = 2;
  }

  bytes job_id = 1;
  repeated StateTimestamp states = 2;
}
